rslib/src/storage/sqlite.rs

```diff
@@ -205,6 +206,36 @@ fn add_regexp_tags_function(db: &Connection) -> rusqlite::Result<()> {
     )
 }
 
+/// eg. get_original_position(c.due, c.data) -> i64
+/// Check if c.data contains the position, if not it's still in c.due
+fn add_get_original_position_function(db: &Connection) -> rusqlite::Result<()> {
```

Comment:
Similar routines above all use extract_ in the name; using it here too would be consistent.

I also I think this routine would be better off focusing on returning position | null. The caller can coalesce the returned null value to card.due if they want to fall back on it. That said, do we really want to be doing that? Can't we just leave cards without an original position unsorted?

Reply:
>The caller can coalesce the returned null value to card.due if they want to fall back on it. That said, do we really want to be doing that? Can't we just leave cards without an original position unsorted?

Unless I'm misunderstanding something, every card will always have an original position. However, the column that stores it will change.
1. User makes card, original position is in the due column
2. User reviews card, due is replaced with a timestamp of next review and original position is pushed to the data column

It makes senses to fallback to the due column since the original position has to be there if it isn't in data column yet.